torchgen/model.py

```diff
@@ -835,7 +835,7 @@ def from_yaml(
         # don't care if it exists or not; make it easier to use this function
         # with other yaml parsers that aren't setting __line__ in the dict
         e.pop("__line__", None)
-        assert not e, f"leftover entries: {e}"
+        # assert not e, f"leftover entries: {e}"
```

Comment:
We can, but we would either need to 
(1) duplicate the logic from `NativeFunctions.from_yaml` or 
(2) propagate the fact that we want to allow additional fields or 
(3) proactively remove the fields before `NativeFunction.from_yaml` calls 

Since `native_functions.yaml` starts with these fields provided (before we translate)

Reply:
We should try to avoid duplication. In the end we only need the NativeFunctions and either backend indices or kernel indices. I believe some refactoring is needed here.